sphinx/ext/autodoc/__init__.py

```diff
@@ -1075,43 +1094,27 @@ def format_args(self, **kwargs: Any) -> str:
     def document_members(self, all_members: bool = False) -> None:
         pass
 
-    def add_directive_header(self, sig: str) -> None:
```

Comment:
I wonder we should keep this as is because some extension inherits this documenter and call this method.

FYI: zopeext is one of the case that inherits `ClassDocumenter`. But it does not call the `add_directive_header()` directly. So it's not a case I said. It lets us know there are extensions that inherit our documenter classes.
https://github.com/sphinx-contrib/zopeext/blob/master/sphinxcontrib/zopeext/autointerface.py

Reply:
I think this is now resolved in my latest revision. Let me know if you see further compatibility issues.